### PR TITLE
improved handling of url in tree modules

### PR DESCRIPTION
**Pull request by sveneberth**
Now the tree widget changes the URL according to the current opened node.
To allow a functionally jump into a node it was necessary to restructure the rebuildPath process of the breadcrumbs. Therefor we need the PR [#141](https://github.com/viur-framework/server/pull/141) of the server which provides the correct usage of `pathToKey`.
This PR is related to Issue #25

### CODE REVIEW

File: widgets/tree.py [hunk 1]
```diff
@@ -5,11 +5,14 @@
...
-class NodeWidget( html5.Div ):
+
+
+class NodeWidget(html5.Div):
 	"""
 		Displays one Node (ie a directory) inside a TreeWidget
 	"""
-	def __init__(self, modul, data, structure, *args, **kwargs ):
+
+	def __init__(self, modul, data, structure, *args, **kwargs):
```
Comment (phorward) [inline widgets/tree.py]: Please change "modul" into "module" here

File: widgets/tree.py [hunk 2]
```diff
@@ -55,20 +58,20 @@ def onDragOver(self, event):
 		try:
 			nodeType, srcKey = event.dataTransfer.getData("Text").split("/")
 		except:
-			return( super(NodeWidget,self).onDragOver(event) )
+			return (super(NodeWidget, self).onDragOver(event))
```
Comment (phorward) [inline widgets/tree.py]: remove brackets

File: widgets/tree.py [hunk 3]
```diff
@@ -55,20 +58,20 @@ def onDragOver(self, event):
...
 		event.preventDefault()
 		event.stopPropagation()
 
 	def onDragLeave(self, event):
 		if "insert_here" in self["class"]:
 			self["class"].remove("insert_here")
-		return( super(NodeWidget, self).onDragLeave(event))
+		return (super(NodeWidget, self).onDragLeave(event))
```
Comment (phorward) [inline widgets/tree.py]: also here!

**Review by phorward — CHANGES_REQUESTED** [review]
Looks good to me so far.
Please change these little things to make it perfect!

File: widgets/tree.py [hunk 4]
```diff
@@ -55,20 +58,20 @@ def onDragOver(self, event):
 		try:
 			nodeType, srcKey = event.dataTransfer.getData("Text").split("/")
```
Comment (sveneberth) [inline widgets/tree.py]: @phorward We assign here two variables, but don't use them. Why?

Comment (phorward) [reply widgets/tree.py]: I don't know ;-)

**Pull request merged**